oeps/architectural-decisions/oep-XXXX-micro-frontend-theming.rst

```diff
@@ -0,0 +1,267 @@
+
+Support
+*******
+
+All configuration variables should be documented with descriptions of what they configure, example values, and indication of whether they're required.
+
+Adding, removing, or modifying a configuration value should be documented in release notes.
+
+Styling
+-------
+
+Styling is a very common - and thus important - method of customization.  It is frequently used to change colors, fonts, background images, and less often to modify spacing or layout.  For the purposes of this OEP, "styling" refers specifically to overriding a micro-frontend's stylesheet.
+
+Implementation Details
+**********************
+
+Open edX micro-frontend stylesheets are generated from a `SASS <https://sass-lang.com/>`_ stylesheet at build-time, and take advantage of the Paragon design system and our "brand" packages to allow operators to substitute alternate styling during the build process, altering the final CSS.
+
+Paragon's styles are based on Bootstrap and make heavy use of SASS variables to define colors, fonts, and spacing.  The stylesheet in a micro-frontend is composed in such a way that when the brand package is included, it acts as an override of any of the defaults specified in Paragon.  We then use an npm module override (see below) to substitute the operator's brand in place of the empty placeholder brand.
+
+Use Cases
+*********
+
+As noted above, styling is the appropriate method of customization to change the visual look and feel of a micro-frontend, up to a point.
+
+Styling and a brand package are also an appropriate way to override images included in the application through the stylesheet, such as images used for ``background-image`` properties.
+
+Support
+*******
+
+We expect that all the CSS utility classes applied to our micro-frontends will be reasonably styleable without creating confusion.  For instance, we should avoid using utility classes like ``bg-white``, which implies a background should be white.  If an operator wants to change a background set to ``bg-white`` to some other color, they're creating ambiguity and confusion in their brand by having to redefine such classes.
+
+Styling is not an appropriate way to "disable" functionality by hiding it, and the platform can't guarantee that the rest of the micro-frontend's layout will react well to a "hole" appearing in the page.
+
+We avoid using semantic CSS classes where possible, in favor of utility classes.  We can't guarantee that semantic CSS classes applied to elements will always be present.
+
+One thing worth noting about the styling customization process is that it must happen at build-time, since we don't ship any SASS to the client, nor is there any run-time SASS compilation in use.  This means that operators who want to run multiple brands on a single instance of the platform must independently deploy the micro-frontend multiple times, once with each brand.
```

Comment:
I've investigated this a bit and I feel if the MFEs avoid using any SCSS that needs theme information, we can build a common scss file that can be rebuild and deployed independently of the MFE. This would mean that you need to pre-build a theme, but can swap it out in real-time. So end-users could potentially be given a light-theme/dark-theme option or a separate service could rebuild the theme and replace it without needing all MFEs to be redeployed.